src/evotorch/core.py

```diff
@@ -1048,7 +1057,7 @@ def get_ray_resource(resource_name: str) -> Any:
 
         # Initialize the Hook instances (and the related status dictionary for the `_after_eval_hook`)
         self._before_eval_hook: Hook = Hook()
-        self._after_eval_hook: Hook = Hook([self._get_best_and_worst])
+        self._after_eval_hook: Hook = Hook()
```

Comment:
Why is this default `after_eval_hook` being quietly removed?

Reply:
We can see this not as a removal, but a slight internal change to do the same thing. Later in the code, `self._get_best_and_worst` is still called in the same moments of execution. The only change is that this method is now explicitly called, instead of via a Hook. If we still did this via Hook, the following would happen:

- The problem object holds a reference to the Hook
- The Hook holds a reference to the problem object's method
- The method, in its context, stores `self`, which is a reference to the problem

So, this is a cycle of references. The default implementations for `Problem.__getstate__`, `Problem.__copy__`, and `Problem.__deepcopy__` introduced in this branch can deal with such cyclic references. However, there could always come a time in which we have a special problem type that we wish to implement, and maybe it will need customized behavior(s) for one or more of these methods for the moments of cloning and/or for serializing. While customizing/overriding these methods, we would always have to remember and write protections against this cyclic reference to avoid things like `RecursionError`. Taking this method outside the Hook simplifies things greatly.